src/reporter/queryContext.py

```diff
@@ -0,0 +1,81 @@
+from exceptions.exceptions import NGSIUsageError, InvalidParameterValue
+from flask import request
+from reporter.reporter import _validate_query_params, _validate_entity
+from translators.factory import translator_for
+import logging
+from .geo_query_handler import handle_geo_query
+import pdb
+
+def queryContext():
+    """
+    See /op/queryContext in API Specification
+    quantumleap.yml
+    """
+    if request.json is None:
+        return 'Discarding query due to lack of request body. ' \
+               'Lost in a redirect maybe?', 400
+
+    if 'entities' not in request.json:
+        return 'Discarding query due to lack of request body ' \
+               'content.', 400
+
+    entity = request.json['entities']
+    attrs = request.json['attributes']
+
+    fiware_s = request.headers.get('fiware-service', None)
+    fiware_sp = request.headers.get('fiware-servicepath', None)
+
+    res = []
+    # Validate entity
+    for et in entity:
+        error = _validate_entity(et)
+        if error:
+            return error, 400
+        entity_type = et["type"]
+        entity_id = et["id"]
+        eid = entity_id.split()
+        entities = None
+        try:
+            with translator_for(fiware_s) as trans:
+                entities = trans.query_last_value(attr_names=attrs,
+                                       entity_type=entity_type,
+                                       entity_ids=eid,
+                                       fiware_service=fiware_s,
+                                       fiware_servicepath=fiware_sp)
+        except NGSIUsageError as e:
+            msg = "Bad Request Error: {}".format(e)
+            logging.getLogger(__name__).error(msg, exc_info=True)
+            return {
+                "error": "{}".format(type(e)),
+                "description": str(e)
+            }, 400
+
+        except InvalidParameterValue as e:
+            msg = "Bad Request Error: {}".format(e)
+            logging.getLogger(__name__).error(msg, exc_info=True)
+            return {
+                "error": "{}".format(type(e)),
+                "description": str(e)
+            }, 422
+
+        except Exception as e:
+            # Temp workaround to debug test_not_found
+            msg = "Something went wrong with QL. Error: {}".format(e)
+            logging.getLogger(__name__).error(msg, exc_info=True)
+            return msg, 500
+
+        if entities:
+            if len(entities) > 1:
```

Comment:
be careful, it may be more than one entity.

Reply:
As per my understanding for 1 entityID and entityType there should be only one entity so only one entity will be returned by the method. Please correct me if I am wrong.

Reply:
you are right, they become more than 1 only after appending them in `res`!